src/functional/ffront/symbol_makers.py

```diff
@@ -147,31 +147,35 @@ def make_symbol_type_from_typing(
             }
 
             # TODO(tehrengruber): print better error when no return type annotation is given
-            return common_types.FunctionType(
+            return ct.FunctionType(
                 args=args, kwargs=kwargs, returns=recursive_make_symbol(return_type)
             )
-        case runtime.Offset:
-            return common_types.OffsetType()
+
+        case type() as t if issubclass(t, common.Dimension):
+            return ct.OffsetType()
 
     raise TypingError(f"'{type_hint}' type is not supported")
 
 
-def make_symbol_type_from_value(value: Any) -> common_types.SymbolType:
-    """Make a symbol type from a Python value."""
-    if isinstance(value, type) or type(value).__module__ == "typing":
-        # we don't have types of types so disallow this
-        raise ValueError("The type of a symbol can not be a type itself.")
-    # TODO(tehrengruber): What we expect here currently is a GTCallable. Maybe
-    #  we should check for the protocol in the future?
-    if hasattr(value, "__gt_type__"):
-        symbol_type = value.__gt_type__()
+def make_symbol_type_from_value(value: Any) -> ct.SymbolType:
+    """Make a symbol node from a Python value."""
+    if isinstance(value, ct.SymbolType):
+        # Temporary hack for type deductions with shifted fields
+        symbol_type = value
```

Comment:
I think this first workaround is only required for the `neighbor_sum` function so if the function is implemented as in my previous suggestion, this can be removed because the `__gt_type__` will cover the case.